### ISSUE TITLE
Managed IAM Policies Fail with any use of `*`

### ISSUE THREAD

**Issue by luckymike** [bridgecrewio/checkov#1037]
**Describe the bug**
In the latest version (1.0.865), it appears that a managed IAM policy will trigger a failure if either of `Resource` or `Action` is set to `*`. This is an overly naive check, as even a policy with `"Effect": "Deny"` will trigger a failure.

I believe this was introduced here: https://github.com/bridgecrewio/checkov/pull/1030

**To Reproduce**
1. Create a managed policy resource with either of `"Resource": "*"` or `"Action": "*"`, with the other scoped narrowly.
2. Run checkcov against this
3. One or more of `CKV_AWS_107`, `CKV_AWS_109`, and `CKV_AWS_111` will fail. 

**Expected behavior**
The policy should pass these checks.

**Additional context**
Even a policy with an explicit Deny all will fail now, e.g.:
```
Check: CKV_AWS_109: "Ensure IAM policies does not allow permissions management without constraints"
	FAILED for resource: AWS::IAM::ManagedPolicy.PlaceholderPolicy
	File: /miscellaneous/legacy_vpc.json:249-274
		249 |   "Resources": {
		250 |     "PlaceholderPolicy": {
		251 |       "Type": "AWS::IAM::ManagedPolicy",
		252 |       "Properties": {
		253 |         "Description": {
		254 |           "Fn::Join": [
		255 |             "",
		256 |             [
		257 |               "Placeholder policy for ",
		258 |               "staging",
		259 |               " legacy VPC"
		260 |             ]
		261 |           ]
		262 |         },
		263 |         "Path": "/placeholder/",
		264 |         "PolicyDocument": {
		265 |           "Version": "2012-10-17",
		266 |           "Statement": [
		267 |             {
		268 |               "Effect": "Deny",
		269 |               "Action": "*",
		270 |               "Resource": "*"
		271 |             }
		272 |           ]
		273 |         }
		274 |       }
```

**Comment by njgibbon** [comment]
@luckymike  Hello, 

Thanks for raising this. 

Over the past month we have been adding the use of Cloudsplaining so that we re-use good work that is already being done out in the community ( https://github.com/salesforce/cloudsplaining ).

CloudFormation has lots of different ways a policy can get in to a Resource so we are working to support the scanning of all of them in an efficient way - hence the ref'd change where we extend to ManagedPolicies.

We need to do more work to rationalise the Cloudsplaining IAM checks and Checkov custom IAM checks and fully and clearly understand the boundary and deltas so we can ensure we like the semantics and adjust where not (i.e probably any type of 'Deny' should always be fine).

I will pick up some of this. Starting with increasing the test coverage so we can isolate which CS checks will still flag this given policy example and go from there.

**Comment by njgibbon** [comment]
Update - I am working on improving some internal docs and testing in Checkov. I have identified why the checks are failing and have initiated some work with Cloudsplaining to get a resolution 🙂 .

**Comment by kmcquade** [comment]
@njgibbon @luckymike - we fixed this on the Cloudsplaining side in 0.4.1. Nick, this should work for Checkov now if you update to the latest version of Cloudsplaining. Let me know if you need anything else with it!

**Comment by schosterbarak** [comment]
Thanks @njgibbon you've bit me to it! You are the best.
And thanks @kmcquade !

I'll bump cloudsplaining version tommorow so it would populate
